### ISSUE TITLE
Potential for interactive use in web frontend

### ISSUE THREAD

**Issue by ekzhang** [lucrae/texlite#56]
Hey @lucrae, I saw this project and was very interested by your idea here! Although the output of LaTeX is really typographically sound, it's super complicated, hard to parse, and sometimes slow. It looks like the language you're designing has a good balance between simplicity, power, and ease of use.

A potential idea on my mind has been making a fast, Markdown-inspired LaTeX variant that compiles to HTML and works right in your browser. This could be used to make a really fast, interactive editor like https://hackmd.io/ or similar. Potentially, it could be written in either JavaScript or Rust, both of which could lead to a really fast renderer and nice editing experience. I also have a personal interest in this, as I use LaTeX to take notes for classes (for example, [this](https://www.ekzhang.com/assets/pdf/Stat_210_Notes.pdf)) and typeset homework.

Just wanted to pitch this idea to you and hear your thoughts, especially if you have any suggestions about design or potential usage cases.

**Comment by lucrae** [comment]
Hey @ekzhang! Sorry for late response (closed-source work has been overtaking my open-source work quite sadly).

I'm so glad you too see the great power in being able to interface the strength of LaTeX with the simplicity of something a lot easier. So, the concept of making that even easier with an in-browser editor is absolutely a fantastic idea. 

Although in best theory one should only really need to render/compile to PDF to get the final document, in reality there's an impulse to update every half minute just to see how beautiful your LaTeX document is, so some fast in-browser editor where rendering is as simple as pressing Ctrl+S and looking to the other half of the screen (rather than switching to a command-line and then opening a PDF) is certainly the ideal setup.

Btw, your stats notes are beautiful, glad to have met another LaTeX fan.
